numpy/core/src/multiarray/shape.c

```diff
@@ -683,25 +672,42 @@ PyArray_SwapAxes(PyArrayObject *ap, int a1, int a2)
                         "bad axis2 argument to swapaxes");
         return NULL;
     }
-    new_axes.ptr = dims;
-    new_axes.len = n;
 
-    for (i = 0; i < n; i++) {
-        if (i == a1) {
-            val = a2;
-        }
-        else if (i == a2) {
-            val = a1;
+    if (a1 == a2) {
+        /* Return a view of the original array */
+        PyObject *ret;
+
+        Py_INCREF(PyArray_DESCR(ap));
+        ret = PyArray_NewFromDescr(Py_TYPE(ap), PyArray_DESCR(ap), n,
```

Comment:
Any idea what the last argument does? It isn't documented... OK, looks like it is needed for `__finalize__`, can be `NULL`.